components/bl-social-login-button-component/src/button.js

```diff
@@ -0,0 +1,49 @@
+import { iconsMap } from './icons-map';
+
+const { cn } = BackendlessUI.CSSUtils;
+
+export function Button({ provider, showButtonIcon, onLogin }) {
+  const handleClick = async () => {
+    async function socialLogin(providerCode, fieldsMappings, scope, options, redirectToPage, callbackUrlDomain) {
+      if (BackendlessUI.DeviceAPI.isMobile()) {
+        const user = await BackendlessUI.DeviceAPI
+          .socialLogin(providerCode, fieldsMappings, scope, options, callbackUrlDomain)
+
+        Backendless.UserService.setCurrentUser(user, true)
+
+        if (redirectToPage) {
+          BackendlessUI.Navigator.goToPage(redirectToPage)
+        }
+      } else {
+        const redirectAfterLoginUrl = redirectToPage
+          ? window.location.href.split('?')[0] + `?page=${ redirectToPage }`
```

Comment:
[URLSearchParams](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams)

Can we use it?

Reply:
Of course we can, but why if it already works.

Reply:
Is it stay on the same page(without reloading) if we did not provide redirectPage?
I could assume that the server return a redirect answer on this request because we provide a redirect parameter. 

And if really need a redirect. Do we save all query parameters? As I see we leave only `page`, but what if we need to pass some other parameters(or keep current)?

Maybe @Valodya could help us with correct behavior there.

Reply:
1) we won't stay on the page because `window.location = await Backendless.UserService.getAuthorizationUrlLink` redirects us to login provider page

2) as far as I know providing the argument `redirectAfterLoginUrl` is required

3) all the query params of the page are already in the PageData so there is no need to keep the current query params, but I assume we have to provide an option to pass extra params which will be applied to a PageData after login (redirection)